PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/lib/common/utils/WebSocketSSLProxy.java

```diff
@@ -0,0 +1,155 @@
+        this.webSocketServerUri = webSocketServerUri;
+        this.id = id;
+        this.password = password;
+    }
+
+    public void start() throws IOException {
+        close();
+        // InetAddress.getByName(null) points to the loopback address (127.0.0.1)
+        serverSocket = new ServerSocket(0, 1, InetAddress.getByName(null));
+        thread = new Thread(() -> {
+            try {
+                for (;;) {
+                    Socket clientSocket = serverSocket.accept();
+                    createWebSocketToSocket(clientSocket);
+                    pipeSocketDataToWebSocket(clientSocket);
+                }
+            } catch (IOException | WebSocketException e) {
+                handleConnectBroken(e);
+            }
+
+        });
+        thread.setName("WebsocketSSLProxy-" + thread.getId());
+        thread.start();
+    }
+
+    public void pipeSocketDataToWebSocket(Socket socket) throws IOException {
+        byte[] buffer = new byte[getBufferSize()];
+        while (true) {
+            int bytesRead = socket.getInputStream().read(buffer);
+            if (bytesRead == -1) {
+                break;
+            }
+
+            webSocket.sendBinary(Arrays.copyOfRange(buffer, 0, bytesRead));
+        }
+    }
+
+    public void createWebSocketToSocket(Socket client) throws IOException, WebSocketException {
+        this.webSocket = new WebSocketFactory().setConnectionTimeout(connectTimeout).createSocket(webSocketServerUri)
+                                               .setUserInfo(this.id, this.password)
+                                               .addListener(new WebSocketAdapter() {
+                                                   @Override
+                                                   public void onBinaryMessage(WebSocket websocket, byte[] bytes) {
+                                                       try {
+                                                           client.getOutputStream().write(bytes);
+                                                       } catch (IOException e) {
+                                                           handleConnectBroken(e);
+                                                       }
+                                                   }
+                                               }).addExtension(WebSocketExtension.PERMESSAGE_DEFLATE).connect();
+
+    }
+
+    public void close() {
+        closeInner();
+    }
+
+    public int getLocalPort() {
+        if (Objects.isNull(serverSocket)) {
+            return 0;
+        }
+        return serverSocket.getLocalPort();
+    }
+
+    public int getBufferSize() {
+        return bufferSize;
+    }
+
+    public void setBufferSize(final int bufferSize) {
+        this.bufferSize = bufferSize;
+    }
+
+    public int getConnectTimeout() {
+        return connectTimeout;
+    }
+
+    public void setConnectTimeout(final int connectTimeout) {
+        this.connectTimeout = connectTimeout;
+    }
```

Comment:
maybe simplified with lombok `@Getter/@Setter`

Reply:
will fix in next PR